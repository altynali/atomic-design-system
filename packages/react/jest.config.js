/** @type {import('jest').Config} */

const config = {
  verbose: true,
  testEnvironment: "jsdom",
  collectCoverageFrom: ["src/**/*.{js,ts, tsx}"],
  roots: ["<rootDir>/src"],
}

module.exports = config
