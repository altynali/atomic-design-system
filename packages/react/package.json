{
  "name": "atomic-design-system-react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/node": "^20.10.0",
    "@types/react": "^18.2.23",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^4.6.0",
    "rollup-plugin-typescript2": "^0.36.0",
    "typescript": "^5.3.2"
  },
  "dependencies": {},
  "scripts": {
    "build": "rollup -c --bundleConfigAsCjs",
    "dev": "yarn build --watch"
  }
}
