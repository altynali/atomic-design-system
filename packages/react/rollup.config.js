import Ts from "rollup-plugin-typescript2"

const config = {
  input: [
    "src/index.ts",
    "src/atoms/Color/index.ts",
    "src/atoms/Text/index.ts",
    "src/atoms/Margin/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/Padding/index.ts",
    "src/atoms/Grid/index.ts",
    "src/atoms/Checkbox/index.ts",
    "src/molecules/Card/index.ts",
    "src/molecules/Select/index.ts",
    "src/molecules/Input/index.ts",
    "src/molecules/CardList/index.ts",
    "src/organisms/Form/index.ts",
    "src/templates/Layout/index.ts",
  ],
  output: [
    {
      dir: "lib",
      format: "esm",
      sourcemap: true,
      preserveModules: true,
    },
  ],
  plugins: [Ts()],
  external: ["react", "atomic-design-system-foundation"],
}

export default config
