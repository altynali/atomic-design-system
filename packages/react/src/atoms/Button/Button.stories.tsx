import React from "react"
import Button from "./Button"
import "atomic-design-system-scss/lib/Button.css"

export default {
  title: "Atoms|Button",
}

export const Common = () => <Button>button</Button>
