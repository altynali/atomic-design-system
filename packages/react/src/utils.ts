export const NumberToPixel = (number: number) => {
  return number.toString() + "px"
}

export const widthOrHeightToPixel = (number: number | undefined) => {
  if (typeof number === "number") {
    return NumberToPixel(number)
  }
  return "fit-content"
}

export const classNames = (cls: string, additional: string[] = []): string => {
  const classes = [cls, ...additional?.filter(Boolean)]

  return classes.join(" ")
}
