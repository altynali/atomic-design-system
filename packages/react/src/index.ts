import { Color } from "./atoms/Color"
import { Text } from "./atoms/Text"
import { Margin } from "./atoms/Margin"
import { Button } from "./atoms/Button"
import { Select } from "./molecules/Select"

export { Color, Text, Margin, Button, Select }

// TODO: import * from './atoms';
