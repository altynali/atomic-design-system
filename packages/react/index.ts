import { Color, Margin, Select, Text } from "./src/index"

export { Color, Text, Margin, Select }

// TODO: import * from './atoms';
