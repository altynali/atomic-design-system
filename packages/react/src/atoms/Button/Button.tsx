import { Sizes } from "atomic-design-system-foundation"
import { SizesType } from "atomic-design-system-foundation/src/types"
import React, { ButtonHTMLAttributes, FC, PropsWithChildren } from "react"

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  hexCode?: string
  size?: SizesType
} & PropsWithChildren

//TODO: complete primary etc
const Button: FC<ButtonProps> = (props) => {
  const { size = Sizes.base, hexCode, children, ...rest } = props

  const className = `atds-button-container atds-button-${size}`

  return (
    <button
      style={{ backgroundColor: hexCode }}
      className={className}
      {...rest}
    >
      {children}
    </button>
  )
}

export default Button
