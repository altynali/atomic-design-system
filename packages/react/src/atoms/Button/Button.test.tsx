import React from "react"
import { render, fireEvent } from "@testing-library/react"
import Button from "./Button"
import { Sizes } from "atomic-design-system-foundation/lib/src"
import "@testing-library/jest-dom"

describe("Button component", () => {
  it("renders button with provided children", () => {
    const { getByText } = render(<Button>Hello</Button>)
    const button = getByText("Hello")
    expect(button).toBeInTheDocument()
    expect(button.tagName).toBe("BUTTON")
  })

  it("applies custom className", () => {
    const { container } = render(<Button className="custom">Click me</Button>)
    const button = container.querySelector("button")
    expect(button).toHaveClass("custom")
  })

  it("applies custom size class", () => {
    const customSize = Sizes.lg

    const { container } = render(<Button size={customSize}>Click me</Button>)
    const button = container.querySelector("button")
    expect(button).toHaveClass(`atds-button-${customSize}`)
  })

  it("applies custom hexCode as backgroundColor style", () => {
    const hexCode = "#FF0000"
    const { container } = render(<Button hexCode={hexCode}>Click me</Button>)
    const button = container.querySelector("button")
    expect(button).toHaveStyle(`background-color: ${hexCode}`)
  })

  it("fires onClick event when clicked", () => {
    const onClickMock = jest.fn()
    const { getByText } = render(
      <Button onClick={onClickMock}>Click me</Button>
    )
    const button = getByText("Click me")
    fireEvent.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  test("snapshot of component", () => {
    const { asFragment } = render(<Button>Hello World</Button>)

    expect(asFragment()).toMatchSnapshot()
  })
})
