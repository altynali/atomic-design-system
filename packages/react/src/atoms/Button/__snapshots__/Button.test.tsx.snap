// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot of component 1`] = `
<DocumentFragment>
  <button
    class=" atds-button-container atds-button-base"
  >
    Hello World
  </button>
</DocumentFragment>
`;
